docs/basics/ExampleTodoList.md

```diff
@@ -89,17 +95,18 @@ function todos(state = [], action) {
       return [
         ...state,
         {
+          id: action.id
           text: action.text,
           completed: false
         }
       ]
     case COMPLETE_TODO:
       return [
-        ...state.slice(0, action.index),
-        Object.assign({}, state[action.index], {
+        ...state.slice(0, action.id),
```

Comment:
This assumes that `id` is also an index, but this may not be true. Instead, I suggest you to use `map()` to iterate over the todos, and return either the original or the copy with different `completed` value depending on its ID. I also suggest you to separate the logic to update a single todo into a separate `todo` reducer like [in my Egghead course](https://egghead.io/lessons/javascript-redux-reducer-composition-with-objects?series=getting-started-with-redux): 

```js
const todo = (state, action) => {
  switch (action.type) {
    case 'ADD_TODO':
      return {
        id: action.id,
        text: action.text,
        completed: false
      };
    case 'TOGGLE_TODO':
      if (state.id !== action.id) {
        return state;
      }

      return {
        ...state,
        completed: !state.completed
      };
    default:
      return state;
  }
};

const todos = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TODO':
      return [
        ...state,
        todo(undefined, action)
      ];
    case 'TOGGLE_TODO':
      return state.map(t =>
        todo(t, action)
      );
    default:
      return state;
  }
};
```